Assets/Scripts/Interactables/CocktailFridge.cs

```diff
@@ -1,13 +1,102 @@
+using UnityEngine;
+using System.Linq;
+
 namespace FMS.TapperRedone.Interactables
 {
-	public class CocktailFridge : InteractableObject
-	{
-		public Inventory.Item itemToSpawn;
-
-		public override void Interact(Characters.PlayerInteraction player)
-		{
-			base.Interact(player);
-			player.CurrentInventory.AddItem(itemToSpawn);
-		}
-	}
+    public class CocktailFridge : InteractableObject
+    {
+        public Inventory.Item[] Ingredients; //array to store colour coded ingredients items as ItemType
+        private SpriteRenderer FridgeSpriteRenderer; // Reference to the SpriteRenderer of the fridge
+
+        private enum ColorCategory //short enum list for the colour window of error
+        {
+            Red,
+            Yellow,
+            Blue,
+            Unknown
+        }
+
+        public void Awake()
+        {
+            if (FridgeSpriteRenderer == null)
+            {
+                //getting the sprite renderer of assigned object prefab 
+                FridgeSpriteRenderer = GetComponent<SpriteRenderer>();
+            }
+        }
+
+        public override void Interact(Characters.PlayerInteraction player)
+        {
+            base.Interact(player);
+           
+
+            if (FridgeSpriteRenderer != null)
+            {
+                Color fridgeColor = FridgeSpriteRenderer.color;
```

Comment:
Repeating comment from elsewhere: 
why do we use the Color of the fridge instead of say setting a ColorCategory on the Fridge?

Reply:
In fact, if you design it that way (Fridge has an enum value for what it can return, or an item definition), it will be more extendable in the future.